### PR TITLE
fix: pick query params with empty values

### LINKED ISSUE #2467: Handle resolving a null field with @http
### Discussed in https://github.com/tailcallhq/tailcall/discussions/2466

<div type='discussions-op-text'>

<sup>Originally posted by **angristan** July 19, 2024</sup>
With this example:

```gql
schema @server(port: 8000) @upstream(baseURL: "http://jsonplaceholder.typicode.com") {
  query: Query
}

type Query {
  users: [User] @http(path: "/users")
}

type User {
  id: Int!
  name: String!
  username: String!
  email: String!
  authorID: String

  posts: [Post]
    @http(
      path: "/posts"
      query: [{ key: "author_ids[]", value: "{{.value.authorID}}" }]
      batchKey: ["author_ids[]"]
    )
}

type Post {
  id: Int!
  title: String!
  body: String!
}
```

If `authorID` is not nil, it works as expected, however if it is nil, it calls `/posts` without any query param, thus returning unrelated posts.

**Expected url**
https://jsonplaceholder.typicode.com/posts?userId=
**Actual url**
https://jsonplaceholder.typicode.com/posts

By default this should be the behaviour

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Issue Reference(s):**  
Fixes #2467 
/claim #2467

### CODE REVIEW

**Review by tusharmath — APPROVED** [review]

**Comment by angristan** [comment]
This doesn't fix the issue for me 🤔  It's still calling the endpoint without the query param for a null value

**Comment by ssddOnTop** [comment]
> This doesn't fix the issue for me 🤔 It's still calling the endpoint without the query param for a null value

can you give a working config pls

**Comment by ssddOnTop** [comment]
@angristan can you join the discord server for faster communication

https://discord.gg/tn2Gtg7s

File: src/core/http/query_encoder.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ impl QueryEncoder {
                 ValueString::String(val) => Some(format!("{}={}", key, val)),
             }
         } else {
-            None
+            Some(key.to_owned())
```
[the remark this answers is not in the document]
Comment (ssddOnTop) [reply src/core/http/query_encoder.rs:22]: hmm got it.. it's because you have different case with List type

**Comment by ssddOnTop** [comment]
@angristan can you try the new changes

**Comment by angristan** [comment]
This fixes it ✅

**Pull request merged**